r1cs/gadgets/std/src/fields/nonnative/params.rs

```diff
@@ -1,91 +1,169 @@
 use crate::fields::nonnative::NonNativeFieldParams;
 
 /// The surfeit used for finding the optimal parameters
-pub(crate) const SURFEIT:u32 = 10; 
+pub(crate) const SURFEIT:u32 = 10;
 
 /// Obtain the parameters from a `ConstraintSystem`'s cache or generate a new one
 #[must_use]
-pub const fn get_params(target_field_size: usize, base_field_size: usize) -> NonNativeFieldParams {
+pub const fn get_params(target_field_prime_length: usize, base_field_prime_length: usize, pseudo_mersenne_const: Option<u64>) -> NonNativeFieldParams {
     let (num_of_limbs, limb_size, _) =
-        find_parameters(base_field_size, target_field_size);
+        find_parameters(target_field_prime_length, base_field_prime_length, pseudo_mersenne_const);
     NonNativeFieldParams {
         num_limbs: num_of_limbs,
         bits_per_limb: limb_size,
     }
 }
 
+const fn ceil_log2(value: u128) -> usize{
+    let mut result = std::mem::size_of::<u128>() * 8 -
+        value.leading_zeros() as usize;
+    if value == 2u128.pow((result - 1) as u32) { result -= 1 }
+    result
+}
+
+// This function estimates the number of constrains for a mul+reduce when the target prime field is
+// pseudo-mersenne and field elements are represented with num_limbs base field elements of size
+// bits_per_limb. The function returns the estimated number of constraints if the security bound for
+// the given parameters holds, None otherwise
+const fn estimate_cost_for_pseudomersenne(target_field_prime_length: usize, base_field_prime_length: usize, pseudo_mersenne_const: u64, bits_per_limb: usize, num_limbs: usize) -> Option<usize> {
```

Comment:
Just a minor formatting thing: I am not a big fan of lines which are longer than 100 characters. Hard to review.

Reply:
Maybe running a "cargo fmt" will help